Lua/Lua.sublime-syntax

```diff
@@ -17,6 +17,13 @@ contexts:
         4: punctuation.definition.parameters.begin.lua
         5: variable.parameter.function.lua
         6: punctuation.definition.parameters.end.lua
+    - match: '\b(function)\s*(\()([^)]*)(\))'
```

Comment:
Would it be possible to match `\b(function)\s*(\()`, and then `push` to handle the rest?

Reply:
What do you mean by push?

Reply:
I took a closer look at your changes. Would changing the regex on line 13 to

```
'\b(function)\b\s*([a-zA-Z_.:]+[.:])?([a-zA-Z_]\w*)?\s*(\()([^)]*)(\))'
```

give us the same desired effect? I think we can drop these additions from line 20 to 26 then.

Reply:
I guess, however it would worth to fix the function names too then:

```'\b(function)\b\s*([a-zA-Z_]\w*[.:])?([a-zA-Z_]\w*)?\s*(\()([^)]*)(\))'```

Only one . or :

And Lua accepts numbers in function scopes too, after the first more strict character.